_posts/2021-10-26-bwc-testing-for-opensearch.markdown

```diff
@@ -0,0 +1,91 @@
+
+An example testcluster initialization is given below. Here the testcluster is named `mycluster` with 4 nodes. Two versions are provided and the first version is the version that gets installed by default when a node starts. It upgrades to the next version using the method mentioned above.
+
+```
+testClusters {
+    "mycluster" {
+        versions = ["7.10.2", project.version]
+        numberOfNodes = 4
+    }
+}
+```
+
+## Types of BWC tests
+
+All the BWC tests in OpenSearch are located inside the `qa` module. The following types of tests are available for BWC:
+
+* **Mixed Cluster:** For a mixed cluster scenario, a test cluster is spun up with multiple nodes of a supported BWC version, then one node upgraded to the current version of OpenSearch using the [`nextNodeToNextVersion`](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java#L408) method resulting in a cluster with a few nodes still on the older version. The corresponding tests are run for the old cluster and for the newly created mixed cluster automating the mixed cluster upgrade path. 
+    * Located in  `qa/mixed-cluster` 
+    * Run using  `./gradlew :qa:mixed-cluster:vM.N.b#mixedClusterTest`
+* **Rolling Upgrade:** For a rolling upgrade scenario, a test cluster is spun up with multiple nodes of a supported BWC version and each node is upgraded one by one to a newer version of OpenSearch using the `nextNodeToNextVersion` method resulting in a fulling upgraded cluster achieved via sequential upgrade. The corresponding tests are run for the old cluster and for each node after it undergoes the upgrade.
+    * Located in `qa/rolling-upgrade`
+    * Run using `./gradlew :qa:rolling-upgrade:vM.N.b#upgradedClusterTest`
+* **Full Cluster Restart:** For a full cluster restart, a test cluster is spun up with multiple nodes of a supported BWC version and all the nodes are upgraded using the [`goToNextVersion`](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java#L388) method resulting in a fully upgrade cluster by restarting. The corresponding tests are run for the old cluster and for the upgraded cluster.
+    * Located in `qa/full-cluster-restart`
+    * Run using `./gradlew :qa:full-cluster-restart:vM.N.b#upgradedClusterTest`
+* **Snapshot Upgrade**: For a snapshot upgrade, two test clusters are created (one with the older version and one with a newer version) and the snapshot restoration is tested between the clusters. Each step tests the creation of repository and snapshot and restoring the snapshot.
+    * Located in `qa/repository-multi-version`
+    * Run using `./gradlew :qa:repository-multi-version:vM.N.b#Step4NewClusterTest`
+
+To run all the backwards compatibility tests:
+ `./gradlew bwcTest`
```

Comment:
Add another line break after the code block - it runs together into one paragraph on the site.